clientv3/watch.go

```diff
@@ -456,11 +462,23 @@ func (w *watchGrpcStream) run() {
 					wc.Send(ws.initReq.toPB())
 				}
 			case pbresp.Canceled:
-				delete(cancelSet, pbresp.WatchId)
-				if ws, ok := w.substreams[pbresp.WatchId]; ok {
-					// signal to stream goroutine to update closingc
-					close(ws.recvc)
-					closing[ws] = struct{}{}
+				if pbresp.CancelReason == pb.WatchResponse_PERMISSION_DENIED {
```

Comment:
does this really need special handling? why isn't the regular cancel=true path enough?